client/src/components/SchedulingPage/AddSchedulingModal.jsx

```diff
@@ -0,0 +1,134 @@
+import { SmallAddIcon } from '@chakra-ui/icons';
+import {
+  Button,
+  Center,
+  FormControl,
+  FormLabel,
+  IconButton,
+  Input,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Select,
+  useDisclosure,
+} from '@chakra-ui/react';
+import useSchedulingForm from 'hooks/useSchedulingForm.jsx';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { useQuery } from 'react-query';
+import { getPartMaterialList } from 'utils/api/parts.js';
+import Loader from 'components/common/Loader.jsx';
+import { getMachines } from 'utils/api/machines.js';
+import TextField from '@material-ui/core/TextField';
+
+const AddSchedulingModal = ({ showButton = false }) => {
+  const { isLoading, data: partsData } = useQuery('parts/materialList', getPartMaterialList);
+  const { isLoading: isLoadingMachine, data: machineData } = useQuery('machine', getMachines);
+
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const {
+    handleQuantityInput,
+    handleCostInput,
+    handleOperatingTime,
+    handleSubmit,
+    isLoadingButton,
+    quantity,
+    cost,
+    operatingTime,
+    partType,
+    machineName,
+  } = useSchedulingForm();
+
+  if (isLoading || isLoadingMachine) {
+    return <Loader />;
+  }
+
+  return (
+    <>
+      {showButton ? (
+        <Center mt={4}>
+          <Button onClick={onOpen}>Add Scheduling</Button>
+        </Center>
+      ) : (
+        <IconButton
+          colorScheme="blue"
+          variant="outline"
+          aria-label="add"
+          float="right"
+          m={2}
+          icon={<SmallAddIcon />}
+          onClick={onOpen}
+        />
+      )}
+      <Modal isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Add a new schedule</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody pb={6}>
+            <FormControl isRequired>
+              <FormLabel>Part Type</FormLabel>
+              <Select placeholder="Select a part" ref={partType}>
```

Comment:
Should we use the list of parts from the part page?

Reply:
i dont think so, i  dont think were scheduling for specific parts

Reply:
Yes, but a specific part type (such as `handle bar`) have different varieties (`handle-bar3000`, `handys`, `gang`) and each variety requires different materials to build it. Refer image below.

![image](https://user-images.githubusercontent.com/60051944/113429731-02c1aa00-93a7-11eb-9c4e-7d542b5e2b04.png)

Reply:
@DavidLiang01 This is actually not true, A part is constructed from a pre defined amount of materials. This is old data, the quality tab has nothing to do with material@

Reply:
the beautiful code I wrote https://github.com/Mackbellemore/soen-390-team07/blob/develop/server/app/validation/parts/partTypes.ts#L8

Reply:
Every handle bar is made from the same amount and type of materials. Only thing that differs is color, name etc

Reply:
so we keep the part category in the scheduling dropdown instead of part name?